common/locks/lock_test.go

```diff
@@ -22,58 +22,112 @@ package locks
 
 import (
 	"context"
+	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/stretchr/testify/suite"
+	"go.uber.org/goleak"
 )
 
-type (
-	LockSuite struct {
-		*require.Assertions // override suite.Suite.Assertions with require.Assertions; this means that s.NotNil(nil) will stop the test, not merely log an error
-		suite.Suite
-	}
-)
-
-func TestLockSuite(t *testing.T) {
-	suite.Run(t, new(LockSuite))
-}
-
-func (s *LockSuite) SetupTest() {
-	s.Assertions = require.New(s.T()) // Have to define our overridden assertions in the test setup. If we did it earlier, s.T() will return nil
-}
-
-func (s *LockSuite) TestBasicLocking() {
+func TestBasicLocking(t *testing.T) {
 	lock := NewMutex()
 	err1 := lock.Lock(context.Background())
-	s.Nil(err1)
+	require.Nil(t, err1)
 	lock.Unlock()
 }
 
-func (s *LockSuite) TestExpiredContext() {
+func TestExpiredContext(t *testing.T) {
 	lock := NewMutex()
 	err1 := lock.Lock(context.Background())
-	s.Nil(err1)
+	require.Nil(t, err1)
 
 	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 	cancel()
 	err2 := lock.Lock(ctx)
-	s.NotNil(err2)
-	s.Equal(err2, ctx.Err())
+	require.NotNil(t, err2)
+	require.Equal(t, err2, ctx.Err())
 
 	lock.Unlock()
 
 	err3 := lock.Lock(context.Background())
-	s.Nil(err3)
+	require.Nil(t, err3)
 	lock.Unlock()
 }
 
 func BenchmarkLock(b *testing.B) {
 	l := NewMutex()
 	ctx := context.Background()
 	for n := 0; n < b.N; n++ {
-		l.Lock(ctx) //nolint:errcheck
-		l.Unlock()  //nolint:staticcheck
+		_ = l.Lock(ctx)
```

Comment:
The `lockInternal` could be optimized by keeping an internal state in an atomic variable. When there's noone holding the lock (state=0) it can be a lightweight operation to lock and return to bypass the goroutine + channel wait.

Reply:
yea, I've got a different PR for making changes.  this is just setting groundwork for "verify that this thing actually works"